src/main/java/com/uber/cadence/migration/MigrationIWorkflowService.java

```diff
@@ -190,6 +190,105 @@ public ListWorkflowExecutionsResponse ListWorkflowExecutions(
     return callOldCluster(listRequest, 0, _listWorkflow);
   }
 
+  @Override
+  public ListOpenWorkflowExecutionsResponse ListOpenWorkflowExecutions(
+      ListOpenWorkflowExecutionsRequest listRequest) throws TException {
+    ListOpenWorkflowExecutionsResponse response;
+    if (listRequest == null) {
+      throw new BadRequestError("List request is null");
+    } else if (!listRequest.isSetDomain()) {
+      throw new BadRequestError("Domain is null");
+    }
+    if (!listRequest.isSetMaximumPageSize()) {
+      listRequest.maximumPageSize = _defaultPageSize;
+    }
+
+    if (!listRequest.isSetNextPageToken() || hasPrefix(listRequest.getNextPageToken(), _marker)) {
+      if (hasPrefix(listRequest.getNextPageToken(), _marker)) {
+        listRequest.setNextPageToken(
+            Arrays.copyOfRange(
+                listRequest.getNextPageToken(),
+                _marker.length,
+                listRequest.getNextPageToken().length));
+      }
+      response = serviceNew.ListOpenWorkflowExecutions(listRequest);
+
+      if (response.getExecutionsSize() < listRequest.getMaximumPageSize()) {
+        int neededPageSize = listRequest.getMaximumPageSize() - response.getExecutionsSize();
+        ListOpenWorkflowExecutionsRequest copiedRequest = new ListOpenWorkflowExecutionsRequest();
+        copiedRequest = listRequest;
```

Comment:
What is the point of doing a shallow copy here? The maximumPageSize of listRequest will also be changed.